docs/source/x_ext.rst

```diff
@@ -557,32 +557,42 @@ Commit interface
   |                    |                        | that the offloaded instruction (``id``) is no longer speculative, will not get killed (e.g. due to misspeculation or an      |
   |                    |                        | exception in a preceding instruction), and is allowed to be committed. If ``commit_valid`` is 1 and ``commit_kill`` is       |
   |                    |                        | 1, then the offloaded instruction (``id``) shall be killed in the |coprocessor| and the |coprocessor| must guarantee that the|
-  |                    |                        | related instruction does/did not change architectural state.                                                                 |
+  |                    |                        | Shall an offloaded instruction be commited or killed?                                                                        |
+  |                    |                        | If ``commit_valid`` is 1 and ``commit_kill`` is 0,  then the core guarantees that the offloaded instruction (``id``) and any |
+  |                    |                        | older (i.e. preceeding) instructions are no longer speculative, will not get killed (e.g. due to misspeculation or an        |
+  |                    |                        | exception in a preceding instruction), and are allowed to be committed.                                                      |
+  |                    |                        | If ``commit_valid`` is 1 and ``commit_kill`` is 1, then the offloaded instruction (``id``) and any newer (i.e. succeeding)   |
+  |                    |                        | instructions shall be killed in the |coprocessor| and the |coprocessor| must guarantee that the related instructions do/did  |
+  |                    |                        | not change architectural state.                                                                                              |
   +--------------------+------------------------+------------------------------------------------------------------------------------------------------------------------------+
 
-The ``commit_valid`` signal will be 1 exactly one ``clk`` cycle for every offloaded instruction by the |coprocessor| (whether accepted or not). The ``id`` value indicates which offloaded
-instruction is allowed to be committed or is supposed to be killed.
+The ``commit_valid`` signal will be 1 exactly one ``clk`` cycle.
+It is not required that a commit transaction is performed for each offloaded instruction individually.
+Instructions can be signalled to be non-speculative or to be killed in batch.
+E.g. signalling the oldest instruction to be killed is equivalent to requesting a flush of the |coprocessor|.
+The first instruction to be considered not-to-be-killed after a commit transaction with ``commit_kill`` as 1,
+is at earliest an instruction with successful issue transaction completing at least one clock cycle later.
```

Comment:
This is a bit vague. What happens to an instruction with issue_valid = 1 at the same time as a kill but issue_ready coming in a cycle later?

Reply:
as discussed, the current assumption towards multi hart scenarios is that the interface is duplicated, and there would be two commit interface with independent ID pools